cpp/src/io/orc/reader_impl.cu

```diff
@@ -796,6 +878,24 @@ void reader::impl::aggregate_child_meta(cudf::detail::host_2dspan<gpu::ColumnDes
       }
       start_row += child_rows;
     }
+
+    // Parent column null mask and null count would be required for child column
+    // to adjust its nullmask.
+    auto type              = out_buffers[parent_col_idx].type.id();
+    auto parent_null_count = static_cast<uint32_t>(out_buffers[parent_col_idx].null_count());
+    auto parent_valid_map  = static_cast<bitmask_type*>(out_buffers[parent_col_idx].null_mask());
+    auto num_rows          = out_buffers[parent_col_idx].size;
+
+    for (uint32_t id = 0; id < p_col.num_children; id++) {
+      const auto child_col_idx = index + id;
+      if (type == type_id::STRUCT) {
+        parent_column_data[child_col_idx] = {parent_valid_map, parent_null_count};
+        // Number of rows in child will remain same as parent in case of struct column
+        num_child_rows[child_col_idx] = num_rows;
+      } else {
+        parent_column_data[child_col_idx] = {nullptr, 0};
+      }
```

Comment:
Does not seem required, std::vector will default-initialize all elements.
```suggestion
      }
```

Reply:
But these vectors are being reused with resize, so I would need this.

Reply:
I'll take another look, thought that resize expands the vector so we have freshly initialized elements.